### PR TITLE
Edit text notebooks

### PR DESCRIPTION
**Pull request by amrzv**
Hi! Mostly made pep8-style formatting in `tutorials/text` notebooks. More specifically:
- fixed typos in `classify_text_with_bert.ipynb`;
- removed unused imports, fixed typos in `image_captioning.ipynb`;
- added f-string formatting in `nmt_with_attention.ipynb `;
- fixed typos, removed redundant f-string formatting in `solve_glue_tasks_using_bert_on_tpu.ipynb`;
- fixed links, removed redundant installation in `text_classification_rnn.ipynb`;
- fixed links, typos and f-strings formatting in `text_generation.ipynb`;
- removed unused imports and installation, fixed typos in `word2vec.ipynb`;
- fixed links, remove unused import in `word_embeddings.ipynb`

### CODE REVIEW

File: site/en/tutorials/text/image_captioning.ipynb [hunk 1]
```diff
@@ -874,17 +868,17 @@
         "        total_loss += t_loss\n",
         "\n",
         "        if batch % 100 == 0:\n",
-        "            print ('Epoch {} Batch {} Loss {:.4f}'.format(\n",
+        "            print('Epoch {} Batch {} Loss {:.4f}'.format(\n",
```
Comment (MarkDaoust) [inline site/en/tutorials/text/image_captioning.ipynb:871]: If you're touching these it's always good to switch `.format` to f-string if you can.

Your `print("string", data)` are good too.

File: site/en/tutorials/text/classify_text_with_bert.ipynb [hunk 2]
```diff
@@ -952,7 +952,7 @@
         "id": "3cOmih754Y_M"
       },
       "source": [
-        "If you want to use your model on [TF Serving](https://www.tensorflow.org/tfx/guide/serving), remember that it will call your SavedModel through one of its named signatures. In Python, you can test them as follows:"
+        "If you want to use your model on [TF Serving](https://www.tensorflow.org/tfx/guide/serving), remember that it will call your SavedModel through one of it's named signatures. In Python, you can test them as follows:"
```
Comment (MarkDaoust) [inline site/en/tutorials/text/classify_text_with_bert.ipynb:955]: Revert this one. Review your "it's vs its". 

 "it's" is "it is" or "it has".
 
  "its" is like "his", "hers", or "theirs".

File: site/en/tutorials/text/nmt_with_attention.ipynb [hunk 3]
```diff
@@ -662,7 +664,7 @@
         "\n",
         "    dec_hidden = enc_hidden\n",
         "\n",
-        "    dec_input = tf.expand_dims([targ_lang.word_index['<start>']] * BATCH_SIZE, 1)\n",
+        "    dec_input = tf.expand_dims([targ_lang.word_index['<start>']]*BATCH_SIZE, 1)\n",
```
Comment (MarkDaoust) [inline site/en/tutorials/text/nmt_with_attention.ipynb:667]: nit: I thought most linters would encourage you to add the spaces here.

File: site/en/tutorials/text/solve_glue_tasks_using_bert_on_tpu.ipynb [hunk 4]
```diff
@@ -1111,7 +1113,7 @@
         "id": "3cOmih754Y_M"
       },
       "source": [
-        "If you want to use your model on [TF Serving](https://www.tensorflow.org/tfx/guide/serving), remember that it will call your SavedModel through one of its named signatures. Notice there are some small differences in the input. In Python, you can test them as follows:"
+        "If you want to use your model on [TF Serving](https://www.tensorflow.org/tfx/guide/serving), remember that it will call your SavedModel through one of it's named signatures. Notice there are some small differences in the input. In Python, you can test them as follows:"
```
Comment (MarkDaoust) [inline site/en/tutorials/text/solve_glue_tasks_using_bert_on_tpu.ipynb:1116]: Here too.

File: site/en/tutorials/text/text_classification_rnn.ipynb [hunk 5]
```diff
@@ -363,21 +364,21 @@
...
         "\n",
-        "  * The main advantage to a bidirectional RNN is that the signal from the beginning of the input doesn't need to be processed all the way through every timestep to affect the output.  \n",
+        "  * The main advantage of a bidirectional RNN is that the signal from the beginning of the input doesn't need to be processed all the way through every timestep to affect the output.  \n",
         "\n",
         "  * The main disadvantage of a bidirectional RNN is that you can't efficiently stream predictions as words are being added to the end.\n",
         "\n",
-        "1. After the RNN has converted the sequence to a single vector the two `layers.Dense` do some final processing, and convert from this vector representation to a single logit as the classification output. \n"
+        "5. After the RNN has converted the sequence to a single vector of the two `layers.Dense` do some final processing, and convert from this vector representation to a single logit as the classification output. \n"
```
Comment (MarkDaoust) [inline site/en/tutorials/text/text_classification_rnn.ipynb:369]: This `of` changes the intended meaning.

The RNN converts the sequence to a vector. The two Dense layers process that vector.

File: site/en/tutorials/text/text_classification_rnn.ipynb [hunk 6]
```diff
@@ -554,13 +555,13 @@
       },
       "outputs": [],
       "source": [
-        "plt.figure(figsize=(16,8))\n",
-        "plt.subplot(1,2,1)\n",
+        "plt.figure(figsize=(16, 8))\n",
+        "plt.subplot(121)\n",
```
Comment (MarkDaoust) [inline site/en/tutorials/text/text_classification_rnn.ipynb:547]: nit: I find these a little too magical. Can we stick with `1, 2, 1`.

File: site/en/tutorials/text/text_classification_rnn.ipynb [hunk 7]
```diff
@@ -603,7 +604,7 @@
         "\n",
         "Here is what the flow of information looks like with `return_sequences=True`:\n",
         "\n",
-        "![layered_bidirectional](images/layered_bidirectional.ipynb)"
+        "![layered_bidirectional](images/layered_bidirectional.png)"
```
Comment (MarkDaoust) [inline site/en/tutorials/text/text_classification_rnn.ipynb:595]: 😬. This has been broken for a long time. Thanks.

File: site/en/tutorials/text/text_classification_rnn.ipynb [hunk 8]
```diff
@@ -698,10 +699,10 @@
       },
       "outputs": [],
       "source": [
-        "plt.figure(figsize=(16,6))\n",
-        "plt.subplot(1,2,1)\n",
+        "plt.figure(figsize=(16, 6))\n",
+        "plt.subplot(121)\n",
```
Comment (MarkDaoust) [inline site/en/tutorials/text/text_classification_rnn.ipynb:691]: Here too.

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
This is a big PR! 

If you send us any more updates it might be easier to manage if you make separate PRs for each file.

I checked everything thoroughly. This is a good set of changes.  There are just a couple of tiny things to fix.

File: site/en/tutorials/text/classify_text_with_bert.ipynb (hunk 2 above)
Comment (amrzv) [reply site/en/tutorials/text/classify_text_with_bert.ipynb:955]: Yes, it has lost _its_ meaning 😐

File: site/en/tutorials/text/text_classification_rnn.ipynb (hunk 5 above)
Comment (amrzv) [reply site/en/tutorials/text/text_classification_rnn.ipynb:369]: Of course, I will return it as before.
P.S: maybe comma should be there...

File: site/en/tutorials/text/text_classification_rnn.ipynb (hunk 6 above)
Comment (amrzv) [reply site/en/tutorials/text/text_classification_rnn.ipynb:547]: Yep, too more syntactic sugar 😉

**Review by MarkDaoust — APPROVED** [review]
Thanks.

**Comment by amrzv** [comment]
There were two merge conflicts, now resolved.

**Comment by MarkDaoust** [comment]
I resolved the conflicts in the merge commit 8402520956f229a0c4aed4b6bcfcad21ed812301.

I'm not 100% sure why this didn't auto-close. I think it saw that there were un-included commits. 

But everything's good now. 

The changes should be published tomorrow.

Thanks.

**Comment by amrzv** [comment]
I saw there were parallel changes in text/ notebooks, maybe this caused conflicts.
By the way, glad to help!

**Comment by yashk2810** [comment]
We rollbacked this change due to failures. @MarkDaoust will roll forward with a fix.

**Comment by lamberta** [comment]
@amrzv Hey, are you able to restore your deleted branch? Seems I'm unable to re-open this PR without it and want to make sure it doesn't get lost in the shuffle. Thanks

**Comment by MarkDaoust** [comment]
Investigating the failures:

1. it looks like switching the `is` to `==` in `solve_glue_tasks_using_bert_on_tpu.ipynb` just exposed the fact that `tfa.metrics.MatthewsCorrelationCoefficient()` was not being actually being called, and was being  used incorrectly (missing a required argument).

```
  if glue_task == 'glue/cola':
    metrics = tfa.metrics.MatthewsCorrelationCoefficient()
  else:
    metrics = tf.keras.metrics.SparseCategoricalAccuracy(
        'accuracy', dtype=tf.float32)
```


2. For `image_captioning.ipynb` it looks like we just got really unlucky (‼️) and had an example generate a caption with exactly 1,3, or 5  words which triggers this bug.

I'll get these fixed and resubmit this PR.

**Comment by amrzv** [comment]
> Investigating the failures:
> 
> 1. it looks like switching the `is` to `==` in `solve_glue_tasks_using_bert_on_tpu.ipynb` just exposed the fact that `tfa.metrics.MatthewsCorrelationCoefficient()` was not being actually being called, and was being  used incorrectly (missing a required argument).
> 
> ```
>   if glue_task == 'glue/cola':
>     metrics = tfa.metrics.MatthewsCorrelationCoefficient()
>   else:
>     metrics = tf.keras.metrics.SparseCategoricalAccuracy(
>         'accuracy', dtype=tf.float32)
> ```
> 
> 1. For `image_captioning.ipynb` it looks like we just got really unlucky (‼️) and had an example generate a caption with exactly 1,3, or 5  words which triggers this bug.
> 
> I'll get these fixed and resubmit this PR.

Hi, @MarkDaoust.
What are the next steps for this PR? Everything was rollbacked. I can create commits again or create separate pr for each .ipynb notebook if needed.

**Comment by MarkDaoust** [comment]
Hi Alekesy, thanks for the ping.

Yeah, looking at the history is confusing. 

It was merged (https://github.com/tensorflow/docs/commit/8402520956f229a0c4aed4b6bcfcad21ed812301). Then Yash rolled it back because of the failures (https://github.com/tensorflow/docs/commit/fd6065cd3902b391661220198e8f90a534b4ea09).

Then I fixed the bugs you discovered (https://github.com/tensorflow/docs/commit/9c23de361bdee3d804acb7132aafb84694e11644, 
https://github.com/tensorflow/docs/commit/24ecfd32e9bb0d729bf908bdfd91960d83946697) and rolled it forward by doing a rollback of the rollback (https://github.com/tensorflow/docs/commit/a7c567d2a730e0e6e6b3741e5be66b1c813549cb)

But our system didn't update the description for the double rollback, so it still says "rollback of {your cl}", but all your changes are merged and should be visible on the site.

Thanks.
